### PR TITLE
Jetpack AI Image: handle enter shortcut on user custom prompt field

### LINKED ISSUE #38102: Jetpack AI Image: trigger a new generation when the "return" action happens and there is a prompt on the custom prompt field
- refer to the design from p1HpG7-ssb-p2

### PR DESCRIPTION
**Pull request by lhkowalski**
Closes #38102.

## Proposed changes:

* Add a `<KeyboardShortcuts />` wrapper on the user custom prompt field to deal with `enter` hits
* When the focus is inside the custom prompt field and the user hits `enter`, we trigger the image generation, unless there is no prompt typed on it (or no content on the post as well, when it's a featured image generation)
* Rename some vars on the `AiImageModal` component so they read less like a function and more like the flags they re

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

* Enable the beta extensions, go to the block editor and add an image block
* Click "Select Image" and then "Generate with AI" to open the image generation tool
* Confirm the UI is showing as expected, with no buggy textareas or misalignements
* Type something on the custom prompt field (example: "a giant cup of coffee") and then hit "enter"
* Confirm the generation will start
* Do the same check for the featured image generator
   * In this case, the "enter" shortcut will work when:
      * the post has content but the custom prompt is empty
      * the post is empty but the user typed a custom prompt
      * both the post and the custom prompt have content
   * It will not work when both the post and the custom prompt are empty (as expected)

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/ai-image/components/ai-image-modal.tsx [hunk 1]
```diff
@@ -138,13 +138,24 @@ export default function AiImageModal( {
...
-									maxLength={ 1000 }
-									rows={ 2 }
-									onChange={ handleUserPromptChange }
-									placeholder={ instructionsPlaceholder }
-								></textarea>
+								<KeyboardShortcuts
+									bindGlobal
+									shortcuts={ {
+										enter: () => {
+											if ( ! disableGenerateButton ) {
```
Comment (dhasilva) [inline projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/ai-image/components/ai-image-modal.tsx]: Unrelated to this PR:
`disableGenerateButton` reads like a function. I would suggest changing it to `generateButtonDisabled`. Same for the other variables.

Comment (dhasilva) [reply projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/ai-image/components/ai-image-modal.tsx]: Pressing Enter once on a page with no content and no instructions do not trigger a generation, but it adds a newline, and then subsequent enters do trigger the generation.

My suggestion here is to consider the button disabled not only if there is no content, but also if there is only whitespace, as sending an empty request causes a `400` error.

Comment (lhkowalski) [reply projects/plugins/jetpack/extensions/plugins/ai-assistant-plugin/components/ai-image/components/ai-image-modal.tsx]: Addressed both your feedbacks, renaming some vars and also trimming the content of the user prompt so remove extra spaces on it, so a field with only new lines can't be submitted.

**Review by dhasilva — APPROVED** [review]
LGTM. Nice!

**Pull request merged**